test/e2e/major_upgrade_v2.go

```diff
@@ -79,7 +80,24 @@ func MajorUpgradeToV2(logger *log.Logger) error {
 		}
 	}
 
-	cancel()
+	// make all nodes in the network restart and ensure that progress is still made
+	for _, node := range testNet.Nodes() {
+		client, err := node.Client()
+		testnet.NoError("failed to get client", err)
+
+		height, err := getHeight(ctx, client, time.Minute)
+		if err != nil {
+			return fmt.Errorf("failed to get height: %w", err)
+		}
+
+		if err := node.Upgrade(latestVersion); err != nil {
+			return fmt.Errorf("failed to restart node: %w", err)
```

Comment:
[optional]

```suggestion
			return fmt.Errorf("failed to upgrade node: %w", err)
```

Reply:
I know the function says upgrade, but it's actually upgrading to the same version, effectively a restart. It could be cleaner if we just have a `Restart` method in testnet that does the same thing

Reply:
+1 restart would be clearer in that case